beacon-chain/rpc/eth/beacon/handlers_pool.go

```diff
@@ -55,39 +55,145 @@ func (s *Server) ListAttestations(w http.ResponseWriter, r *http.Request) {
-			a, ok := att.(*eth.Attestation)
-			if ok {
-				filteredAtts = append(filteredAtts, structs.AttFromConsensus(a))
-			} else {
-				httputil.HandleError(w, fmt.Sprintf("unable to convert attestations of type %T", att), http.StatusInternalServerError)
+	attestations := s.AttestationsPool.AggregatedAttestations()
+	unaggAtts, err := s.AttestationsPool.UnaggregatedAttestations()
+	if err != nil {
+		httputil.HandleError(w, "Could not get unaggregated attestations: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	attestations = append(attestations, unaggAtts...)
+
+	var firstVersion int
+	versionSet := false
+	var attsData json.RawMessage
+
+	isEmptyReq := rawSlot == "" && rawCommitteeIndex == ""
+	if isEmptyReq {
+		allAtts := make([]interface{}, len(attestations))
+		for i, att := range attestations {
+			switch a := att.(type) {
+			case *eth.AttestationElectra:
+				allAtts[i] = structs.AttElectraFromConsensus(a)
+				if !versionSet {
+					firstVersion = a.Version()
+					versionSet = true
+				}
+			case *eth.Attestation:
+				allAtts[i] = structs.AttFromConsensus(a)
+				if !versionSet {
+					firstVersion = a.Version()
+					versionSet = true
+				}
+			default:
+				httputil.HandleError(w, fmt.Sprintf("Unable to convert attestation of type %T", att), http.StatusInternalServerError)
 				return
 			}
 		}
+		attsData, err = json.Marshal(allAtts)
+		if err != nil {
+			httputil.HandleError(w, "Could not marshal attestations: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+	} else {
+		bothDefined := rawSlot != "" && rawCommitteeIndex != ""
+		filteredAtts := make([]interface{}, 0, len(attestations))
+		for _, att := range attestations {
+			committeeIndexMatch := rawCommitteeIndex != "" && att.GetData().CommitteeIndex == primitives.CommitteeIndex(committeeIndex)
+			slotMatch := rawSlot != "" && att.GetData().Slot == primitives.Slot(slot)
+			shouldAppend := (bothDefined && committeeIndexMatch && slotMatch) || (!bothDefined && (committeeIndexMatch || slotMatch))
+
+			if shouldAppend {
+				switch a := att.(type) {
+				case *eth.AttestationElectra:
+					filteredAtts = append(filteredAtts, structs.AttElectraFromConsensus(a))
+					if !versionSet {
+						firstVersion = a.Version()
+						versionSet = true
+					}
+				case *eth.Attestation:
+					filteredAtts = append(filteredAtts, structs.AttFromConsensus(a))
+					if !versionSet {
+						firstVersion = a.Version()
+						versionSet = true
+					}
+				default:
+					httputil.HandleError(w, fmt.Sprintf("Unable to convert attestation of type %T", att), http.StatusInternalServerError)
+					return
+				}
+			}
+		}
+		attsData, err = json.Marshal(filteredAtts)
+		if err != nil {
+			httputil.HandleError(w, "Could not marshal attestations: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
 	}
-	httputil.WriteJson(w, &structs.ListAttestationsResponse{Data: filteredAtts})
+	w.Header().Set(api.VersionHeader, version.String(firstVersion))
```

Comment:
Same idea as for slashings: we should use the head state or current slot to determine the version and only return attestations for that version.